### PR TITLE
Extending @typescript-eslint/* range

### LINKED ISSUE #332: Update range of compatibility for @typescript-eslint's plugin and parser
#### Issue
`typescript-eslint` recently released [v7.0.1](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v7.0.1) and `eslint-config-airbnb-typescript` does not support it yet

**Current version support:**
```
    "@typescript-eslint/eslint-plugin": "^5.13.0 || ^6.0.0",
    "@typescript-eslint/parser": "^5.0.0 || ^6.0.0",
```

**Desired version support:**
To also include ^7.0.1

### PR DESCRIPTION
**Pull request by aequasi**
Fixes #332

### CODE REVIEW

**Comment by heath-freenome** [comment]
@iamturns Can you please merge this and release? My company is blocked on this. Thanks!!!

**Comment by heath-freenome** [comment]
@aequasi Also consider updating the `README.md` file with an example of using `7.0.1`

File: package.json [hunk 1]
```diff
@@ -26,14 +26,14 @@
...
     "eslint": "^7.32.0 || ^8.2.0",
     "eslint-plugin-import": "^2.25.3"
   },
   "devDependencies": {
-    "@typescript-eslint/eslint-plugin": "^5.13.0 || ^6.0.0",
-    "@typescript-eslint/parser": "^5.0.0 || ^6.0.0",
+    "@typescript-eslint/eslint-plugin": "^5.13.0 || ^6.0.0 || ^7.0.1",
+    "@typescript-eslint/parser": "^5.0.0 || ^6.0.0 || ^7.0.1",
     "doctoc": "2.1.0",
     "eslint": "8.11.0",
```
Comment (RobinTail) [inline package.json:38]: FYI, v7 requires higher `eslint` version

https://github.com/typescript-eslint/typescript-eslint/blob/98aa4f3d3c1522b0670dcaf9e614c8e128edf127/packages/parser/package.json#L51

**Comment by RobinTail** [comment]
@iamturns , please unblock

**Comment by TobiTenno** [comment]
fwiw, this is my "working" overrides in my pacakge.json

```json
{
  "overrides": {
    "eslint-config-airbnb-typescript": {
      "eslint": "^8.57.0",
      "@typescript-eslint/parser": "^7.0.2",
      "@typescript-eslint/eslint-plugin": "^7.0.2"
    }
  }
}
```

**Comment by slifty** [comment]
Given how long this is taking to merge, might it make sense to make the dependency range even broader than the PR's?

e.g. `>5.0.0` which would mean *any* version greater than 5?  That way when v8 eventually comes out it won't require manual intervention.

**Comment by heath-freenome** [comment]
@iamturns I can imagine you are busy... Would you please merge this?

**Comment by DePasqualeOrg** [comment]
Given that this package gets over 2 million downloads a week on NPM, it should really be maintained by someone who has the time to do so. Paging @airbnb @hshoff

**Comment by iamturns** [comment]
Thank you for the PR @aequasi !

This change also required a bump of eslint and an update to the README.

Due to the high demand of this change, instead of requesting changes, I implemented the changes in a different PR: #334 

Thanks again!

**Comment by iamturns** [comment]
Appreciate all your patience, life has become busy for me. I'd be very happy to handover to Airbnb to maintain - please reach out!

**Pull request closed without merging**